### PR TITLE
Add how checkout block processes an order to the repo docs

### PR DESCRIPTION
**Pull request by alexflorisca**
This PR adds the "How The Checkout Block Processes An Order" P2 (pdFofs-hi-p2) to the repo docs based on @beaulebens suggestion (pdFofs-hi-p2#comment-306)

### CODE REVIEW

File: docs/third-party-developers/extensibility/README.md [hunk 1]
```diff
@@ -37,14 +37,15 @@ These documents are all dealing with extensibility in the various WooCommerce Bl
...
-| [DOM Events](./checkout-block/dom-events.md)                            | A list of DOM Events used by some blocks to communicate between them and with other parts of WooCommerce.         |
-| [Filter Registry](../../../packages/checkout/filter-registry/README.md) | The filter registry allows callbacks to be registered to manipulate certain values.                               |
+| Document                                                                                         | Description                                                                                                       |
+| ------------------------------------------------------------------------------------------------ | ----------------------------------------------------------------------------------------------------------------- |
+| [How the Checkout Block processes an order](./checkout-block/how-checkout-processes-an-order.md) | The `IntegrationInterface` class and how to use it to register scripts, styles, and data with WooCommerce Blocks. |
```
Comment (tarunvijwani) [inline docs/third-party-developers/extensibility/README.md]:
```suggestion
| [How the Checkout Block processes an order](./checkout-block/how-checkout-processes-an-order.md) | The detailed inner workings of the Checkout Flow. |
```

I think you missed updating the description here. 🙂

Comment (alexflorisca) [reply docs/third-party-developers/extensibility/README.md]: good spot, thanks Tarun!

**Review by tarunvijwani — CHANGES_REQUESTED** [review]
Thank you for working on this! This is very useful document. It helped me learn the whole checkout flow. 

I feel the tone and sentence framing used in the document could be modified better. It is good for our internal p2s. However, for the product documents, it should sound formal.


Example: 
This lives [here](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/assets/js/base/context/providers/cart-checkout/checkout-provider.js), and wraps all the contexts mentioned above around the CheckoutProcessor component.

Can be converted to:

The [Checkout Provider](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/assets/js/base/context/providers/cart-checkout/checkout-provider.js)  wraps all the contexts mentioned above around the CheckoutProcessor component.

`This is what happens when a user places an order:` can be converted to `Below is the complete checkout flow:`

I'd love to pair with you once you're back from AFK and we can work on it together. It would be a great learning for me. 🙂

**Review by tarunvijwani — APPROVED** [review]
The doc looks good! Thank you so much for publishing it. 👍

**Pull request merged**